tests/rptest/tests/describe_topics_test.py

```diff
@@ -167,38 +148,86 @@ def test_describe_topics_with_documentation_and_types(self):
+        kcl = KCL(self.redpanda)
+        output = kcl.describe_topic(tp_spec.name,
+                                    with_docs=True,
+                                    with_types=True).lower().splitlines()
+
+        # The output format from topic describe has the following structure
+        #
+        # - A table section with descriptions for the property name, type, value, and source
+        # - An empty line
+        # - Three lines for each property, called the documentation section:
+        #     1. Property name with a colon (e.g., cleanup.policy:)
+        #     2. Property documentation string
+        #     3. An empty line
+
+        # First check redpanda.datapolicy in the property table because the value is a special case.
+        # The extra space is necessary so the datapolicy lines in the documentation section are not removed.
+        copro_lines = [
+            line for line in output if line.startswith('redpanda.datapolicy ')
+        ]
+        self.logger.debug(f"Copro lines {copro_lines}")
+        assert len(copro_lines) == 1
+        copro_match = re.match(r"^.*default_config$", copro_lines[0])
+        assert copro_match is not None
+
+        all_other_lines = [
+            line for line in output
+            if not line.startswith('redpanda.datapolicy ')
+        ]
+        property_re = re.compile(
+            r"^(?P<name>[a-z.]+?)\s+(?P<type>[a-z]+?)\s+(?P<value>\S+?)\s+(?P<src>[a-z_]+?)$"
+        )
+
+        for line in all_other_lines:
+            self.logger.debug(f"Property line {line}")
+            prop_match = property_re.match(line)
+
+            if prop_match is None:
+                break
+
+            name = prop_match.group("name")
+            config_type = prop_match.group("type")
+            value = prop_match.group("value")
+            source_type = prop_match.group("src")
+            self.logger.debug(
+                f"name: {name}, type: {config_type}, value: {value}, src: {source_type}"
+            )
+            assert name in properties
             prop = properties[name]
             assert config_type == prop.config_type
             assert value == prop.value
             assert source_type == prop.source_type
 
-        # The output format follows:
-        # line 1 is empty
-        # line 2 is the property name
-        # line 3 is the property docstring
-        prop = None
-        for i in range(0, len(documentation_lines), 3):
-            # The property names have a colon at the end
-            prop_match = re.match(r"^(?P<name>[A-Za-z.]+?):$", line[1])
-            if prop_match is not None:
-                name = prop_match.group("name").lower()
-                assert name in properties.keys()
-                prop = properties[name]
-                assert line[2].lower() == prop.doc_string
+        # The first empty line is where the table ends and the doc section begins
+        empty_line_pos = all_other_lines.index("")
```

Comment:
`list::index` throws `ValueError` if the item isn't found. Also now Denis suggestion isn't working because we aren't starting parsing where we left off in the logic above at the `break` statement. I know I suggested a for loop but I think we can keep it and a simple solution would be to save the index of where you are iterating in the `all_other_lines` array before you `break`.
```
last_idx = None
for idx, line in enumerate(all_other_lines):
        if prop_match is None:
            last_idx = idx
            break
...
assert last_idx is not None
doc_lines = all_other_lines[last_idx:]
assert doc_lines[0] == "", "Expected empty separator"
all_other_lines = all_other_lines[1:]
...
```
Your iterator solution was kind of the same, I'm not super versed in python a quick google search found `next(i, None)` will avoid throwing and return `None` if there's no way to increment the iterator, resolving my previous concerns. I'll leave the implementation decisions up to you.

Reply:
I personally like `while` loops and `next` but I agree that using `for` loops is easier to read so I'll go with that design so future devs will have an easier experience reading the test.

Reply:
I actually like the iterator approach here too. Not suggesting you change anything though. Maybe in your original commits the while loop would have been:
```
>>> a = [1,2,3,4]
>>> itr = iter(a)
>>> while itr is not None and your_stop_condition(itr):
...       print(*itr) # replace with your logic
...       itr = next(itr, None)
... 
1 2 3 4
```
Kind of the same as using an index but yeah I agree an iterator expresses the intention better